gui/src/app/prototypes/AnalysisPyPrototype/analysisPyWorker.ts

```diff
@@ -0,0 +1,91 @@
+import { PyodideInterface, loadPyodide } from "pyodide";
+
+let pyodide: PyodideInterface | null = null;
+const loadPyodideInstance = async () => {
+  if (pyodide === null) {
+    const p = await loadPyodide({
+      indexURL: "https://cdn.jsdelivr.net/pyodide/v0.26.1/full",
+      stdout: (x: string) => {
+        self.postMessage({ type: "stdout", data: x });
+      },
+      stderr: (x: string) => {
+        self.postMessage({ type: "stderr", data: x });
+      },
+    });
+    pyodide = p;
+    await pyodide.loadPackage(["numpy", "matplotlib"]);
+    return pyodide;
```

Comment:
Note for later: I was able to get arviz working by adding this code before this return:

```js
    const micropip = pyodide.pyimport("micropip");
    await micropip.install("arviz<0.18");
```


Then this example runs!

```python
import matplotlib.pyplot as plt

import arviz as az

az.style.use("arviz-doc")

data = az.load_arviz_data("non_centered_eight")
az.plot_trace(data, var_names=("tau", "mu"))

plt.show()
```

Reply:
Very cool. I have implemented this conditionally on whether the script has the word "arviz" in it.